docs/app-dev/app-architecture.md

```diff
@@ -11,7 +11,14 @@ The following diagram provides a superb example:
 
 ![](../imgs/cosmos-tendermint-stack-4k.jpg)
 
-The end-user application here is the Cosmos Voyager, at the bottom left.
+We distinguish here between two forms of "application". The first is the
+end-user application, like a desktop-based wallet app that a user downloads,
+which is where the user actually interacts with the system. The other is the 
+ABCI application, which is the logic that actually runs on the blockchain.
+Transactions sent by an end-user application are ultimately processed by the ABCI
+application after being committed by the Tendermint consensus.
+
+The end-user application in this diagram the Cosmos Voyager, at the bottom left.
```

Comment:
Missig "is"? "in this diagram the Cosmos Voyager" ==> "in this diagram **is** the Cosmos Voyager"?